source/slang/slang.cpp

```diff
@@ -4187,6 +4346,37 @@ SpecializedComponentType::SpecializedComponentType(
     collector.visitSpecialized(this);
 }
 
+SlangResult SpecializedComponentType::computeDependencyBasedHash(slang::Checksum* outHashCode)
+{
+    slang::Checksum hashCode;
+    MD5HashGen hashGen;
+    MD5Context context;
+    hashGen.init(&context);
+
+    auto fileDependencies = getFilePathDependencies();
+    for (auto& file : fileDependencies)
+    {
+        hashGen.update(&context, file);
+    }
```

Comment:
As per an earlier comment, we should only need *one* site in the code where we loop over the file-path dependencies for the root `ComponentType` for which a hash is being requested. This logic shouldn't need to be part of the recursive worker routine.

Reply:
Moved up to `CompositeComponentType`.

Reply:
Should be in ComponentType.

Reply:
Fixed locally, will be in next push